### ISSUE TITLE
[Linux] Converting output value on Script Canvas Node into Variable Node and back to reference crashes the Editor

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#9335]
**Describe the bug**
If a variable is assigned to a Script Canvas Node's output value, then is converted into a Variable Node, and back again into reference, the Editor crashes. It should be noted that a crash.log file is not generated when this occurs, and no notable information is registered to the Editor.log file.

It should also be noted that this issue does not occur on Windows.

**Steps to reproduce**
1. Go to Tools > Script Canvas.
2. Create a new script (File > New Script).
3. Click the "Create Variable" button in the top-right corner, and select the Number type from the dropdown list.
5. Add a Length (Math/Vector3 category) node to the graph. 
6. Drag and drop variable created in step 3 onto the Result output value in the Length node.
7. Right-click the Result value pin, and select "Convert to Variable Node" from the dropdown menu.
8. Right-click the created "Set <Variable Name>" node and select "Convert to References" from the dropdown menu.

**Expected behavior**
The Set <Variable Name> is converted back to the value reference.

**Actual behavior**
The Editor crashes.

**Video**

https://user-images.githubusercontent.com/86952082/166228257-31793ae5-eed7-4756-be48-7ae9151cd40b.mp4

**Found in Branch**
Stabilization_2205 (a2a15da)

**Desktop**
 - Device: PC
 - OS: Linux
 - Version: Ubuntu 20.04.3 LTS
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by carlitosan** [comment]
Is there a callstack?

**Comment by LB-ArturZieba** [comment]
As mentioned in the description, the error.log file is not generated when this occurs, and no notable information is registered to the Editor.log file.

**Comment by spham-amzn** [comment]
Here is one call stack
```
ScriptCanvas::Nodes::Core::SetVariableNode::GetScopedVariableId() const (/src/github/o3de/Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Core/SetVariable.cpp:365)
ScriptCanvas::Nodes::Core::SetVariableNode::GetGraphVariables() const (/src/github/o3de/Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Core/SetVariable.cpp:250)
AZ::AttributeMemberFunction<AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator> (ScriptCanvas::Nodes::Core::SetVariableNode::*)() const>::Invoke(void*) (/src/github/o3de/Code/Framework/AzCore/AzCore/RTTI/ReflectContext.h:361)
bool AZ::Internal::AttributeRead<AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator>, AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator>, void*&>(AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator>&, AZ::Attribute*, void*&) (/src/github/o3de/Code/Framework/AzCore/AzCore/RTTI/AttributeReader.h:31)
bool AZ::Internal::AttributeReader<AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator>, AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator> >::Read<void*&>(AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator>&, AZ::Attribute*, void*&) (/src/github/o3de/Code/Framework/AzCore/AzCore/RTTI/AttributeReader.h:141)
bool AZ::AttributeInvoker<void>::Read<AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator>, AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator> >(AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator>&) (/src/github/o3de/Code/Framework/AzCore/AzCore/RTTI/AttributeReader.h:169)
AzToolsFramework::GenericComboBoxHandler<ScriptCanvas::VariableId>::ConsumeAttribute(AzToolsFramework::GenericComboBoxCtrlBase*, unsigned int, AZ::AttributeReader*, char const*) (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/GenericComboBoxCtrl.inl:267)
non-virtual thunk to AzToolsFramework::GenericComboBoxHandler<ScriptCanvas::VariableId>::ConsumeAttribute(AzToolsFramework::GenericComboBoxCtrlBase*, unsigned int, AZ::AttributeReader*, char const*) (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/GenericComboBoxCtrl.h:0)
auto AzToolsFramework::PropertyHandler_Internal<AzToolsFramework::GenericComboBoxCtrlBase>::ConsumeAttributes_Internal(QWidget*, AzToolsFramework::InstanceDataNode*)::'lambda'(auto const&, char const*)::operator()<AZStd::vector<AZStd::pair<unsigned int, AZ::Attribute*>, AZStd::allocator> >(auto const&, char const*) const (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyEditorAPI_Internals_Impl.h:56)
AzToolsFramework::PropertyHandler_Internal<AzToolsFramework::GenericComboBoxCtrlBase>::ConsumeAttributes_Internal(QWidget*, AzToolsFramework::InstanceDataNode*) (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyEditorAPI_Internals_Impl.h:81)
AzToolsFramework::ReflectedPropertyEditor::Impl::CreateEditorWidget(AzToolsFramework::PropertyRowWidget*) (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ReflectedPropertyEditor.cpp:645)
AzToolsFramework::ReflectedPropertyEditor::Impl::UpdateExpansionState() (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ReflectedPropertyEditor.cpp:622)
AzToolsFramework::ReflectedPropertyEditor::InvalidateAll(char const*) (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ReflectedPropertyEditor.cpp:896)
AzToolsFramework::ReflectedPropertyEditor::DoRefresh() (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ReflectedPropertyEditor.cpp:2316)
AzToolsFramework::ReflectedPropertyEditor::qt_static_metacall(QObject*, QMetaObject::Call, int, void**) (/src/github/o3de/build/linux_clang/Code/Framework/AzToolsFramework/AzToolsFramework_autogen/include_debug/UI/PropertyEditor/moc_ReflectedPropertyEditor.cpp:110)
QObject::event(QEvent*) (@QObject::event(QEvent*):186)
QWidget::event(QEvent*) (@QWidget::event(QEvent*):143)
QFrame::event(QEvent*) (@QFrame::event(QEvent*):14)
QApplicationPrivate::notify_helper(QObject*, QEvent*) (@QApplicationPrivate::notify_helper(QObject*, QEvent*):42)
QApplication::notify(QObject*, QEvent*) (@QApplication::notify(QObject*, QEvent*):103)
```

**Comment by spham-amzn** [comment]
Here is a different call stack from the same repo:

```
ScriptCanvas::Nodes::Core::SetVariableNode::GetScopedVariableId() const (/src/github/o3de/Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Core/SetVariable.cpp:365)
ScriptCanvas::Nodes::Core::SetVariableNode::GetGraphVariables() const (/src/github/o3de/Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Core/SetVariable.cpp:250)
AZ::AttributeMemberFunction<AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator> (ScriptCanvas::Nodes::Core::SetVariableNode::*)() const>::Invoke(void*) (/src/github/o3de/Code/Framework/AzCore/AzCore/RTTI/ReflectContext.h:361)
bool AZ::Internal::AttributeRead<AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator>, AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator>, void*&>(AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator>&, AZ::Attribute*, void*&) (/src/github/o3de/Code/Framework/AzCore/AzCore/RTTI/AttributeReader.h:31)
bool AZ::Internal::AttributeReader<AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator>, AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator> >::Read<void*&>(AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator>&, AZ::Attribute*, void*&) (/src/github/o3de/Code/Framework/AzCore/AzCore/RTTI/AttributeReader.h:141)
bool AZ::AttributeInvoker<void>::Read<AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator>, AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator> >(AZStd::vector<AZStd::pair<ScriptCanvas::VariableId, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> >, AZStd::allocator>&) (/src/github/o3de/Code/Framework/AzCore/AzCore/RTTI/AttributeReader.h:169)
AzToolsFramework::GenericComboBoxHandler<ScriptCanvas::VariableId>::ConsumeAttribute(AzToolsFramework::GenericComboBoxCtrlBase*, unsigned int, AZ::AttributeReader*, char const*) (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/GenericComboBoxCtrl.inl:267)
non-virtual thunk to AzToolsFramework::GenericComboBoxHandler<ScriptCanvas::VariableId>::ConsumeAttribute(AzToolsFramework::GenericComboBoxCtrlBase*, unsigned int, AZ::AttributeReader*, char const*) (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/GenericComboBoxCtrl.h:0)
auto AzToolsFramework::PropertyHandler_Internal<AzToolsFramework::GenericComboBoxCtrlBase>::ConsumeAttributes_Internal(QWidget*, AzToolsFramework::InstanceDataNode*)::'lambda'(auto const&, char const*)::operator()<AZStd::vector<AZStd::pair<unsigned int, AZ::Attribute*>, AZStd::allocator> >(auto const&, char const*) const (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyEditorAPI_Internals_Impl.h:56)
AzToolsFramework::PropertyHandler_Internal<AzToolsFramework::GenericComboBoxCtrlBase>::ConsumeAttributes_Internal(QWidget*, AzToolsFramework::InstanceDataNode*) (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyEditorAPI_Internals_Impl.h:81)
AzToolsFramework::ReflectedPropertyEditor::Impl::CreateEditorWidget(AzToolsFramework::PropertyRowWidget*) (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ReflectedPropertyEditor.cpp:645)
AzToolsFramework::ReflectedPropertyEditor::Impl::UpdateExpansionState() (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ReflectedPropertyEditor.cpp:622)
AzToolsFramework::ReflectedPropertyEditor::InvalidateAll(char const*) (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ReflectedPropertyEditor.cpp:896)
AzToolsFramework::ReflectedPropertyEditor::DoRefresh() (/src/github/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ReflectedPropertyEditor.cpp:2316)
AzToolsFramework::ReflectedPropertyEditor::qt_static_metacall(QObject*, QMetaObject::Call, int, void**) (/src/github/o3de/build/linux_clang/Code/Framework/AzToolsFramework/AzToolsFramework_autogen/include_debug/UI/PropertyEditor/moc_ReflectedPropertyEditor.cpp:110)
QObject::event(QEvent*) (@QObject::event(QEvent*):186)
QWidget::event(QEvent*) (@QWidget::event(QEvent*):143)
QFrame::event(QEvent*) (@QFrame::event(QEvent*):14)
QApplicationPrivate::notify_helper(QObject*, QEvent*) (@QApplicationPrivate::notify_helper(QObject*, QEvent*):42)
QApplication::notify(QObject*, QEvent*) (@QApplication::notify(QObject*, QEvent*):103)
```

**Comment by spham-amzn** [comment]
At first glance, it appears like a race condition that occurs when ReflectedPropertyEditor::DoRefresh is invoked in response to any action that causes a Node deletion

**Comment by LB-KacperKapusta** [comment]
Verified the issue as fixed using Stabilization/2210 (99608ff) build.
